Major revision to Editing Moves (Gen IV) guide (#139)
ds-pokemon-hacking/ds-pokemon-hacking.github.io: docs/generation-iv/guides/editing_moves/editing_moves.md

Reviewer: ```suggestion
- `133` (*Unused, but does have the same trainer move selection AI routines associated to it as healing moves such as Synthesis*)
```

Reviewer: ```suggestion
- `134` (*Unused, but does have the same trainer move selection AI routines associated to it as healing moves such as Synthesis*)
```

| **Move Data**              | `/a/0/1/1` | `/poketool/waza/pl_waza_tbl.narc` | `/poketool/waza/waza_tbl.narc` |
| **Move Scripts**           | `/a/0/0/0` | `/battle/skill/waza_seq.narc`     | `/battle/skill/waza_seq.narc`  |
| **Battle Effect Scripts**  | `/a/0/3/0` | `/battle/skill/be_seq.narc`       | `/battle/skill/be_seq.narc`    |
| **Battle Subscripts**      | `/a/0/0/1` | `/battle/skill/sub_seq.narc`      | `/battle/skill/sub_seq.narc`   | 
| **Move Animation Scripts** | `/a/0/1/0` | `/wazaeffect/we.arc`              | `/wazaeffect/we.arc`           |
| **Continuous Animations**  | `/a/0/6/1` | `/wazaeffect/we_sub.narc`         | `/wazaeffect/we_sub.narc`      |

:::info
Terms are subject to variations to reflect the constantly changing community-defined naming conventions and knowledge of how the game works. For instance, the Diamond/Pearl/Platinum file `be_seq.narc` can be interpreted as "**b**attle **e**ffect **seq**uence," with each individual file in the NARC being a "battle effect sequence script," and has been previously referred to as "move effect scripts" and such.
:::

### Move Data

> Every move has their own set of **move data** that include the move's name, type, category, power, accuracy, PP, battle effect, battle effect chance, priority, target range, contest effect, and other special flags. Most of this data can be edited in **DSPRE's Move Editor**. 

Reviewer: "battle effect chance" --> since this applies for indirect secondary effects only, maybe word differently?

See [Basic Move Data](#basic-move-data) for more details and guides on how to edit move data.

### Move Scripts

> Every move is uniquely associated with a **move script**, both sharing the same ID. In other words, a uniquely identified move script is associated strictly with a specific move and may not shared. 

Reviewer: ```suggestion
Every move is uniquely associated with a **move script**, both sharing the same ID. In other words, a uniquely identified move script is associated strictly with a specific move and may not be shared. 
```

Because most move scripts lack significant complexity or elements, this guide will not have a dedicated section for move scripts specifically. However, the [Battle Effects](#battle-effects) section provides relevant information as move scripts and battle effect scripts make use of the same instructions and logic. Additionally, the [Trapping (Binding) Moves](#trapping-binding-moves) section briefly mentions move scripts due to its relevance for that topic. 

Move scripts for each move can be found in the following decompilation projects:
- [PokePlatinum/res/moves/](https://github.com/pret/pokeplatinum/tree/8b6fa504086925c957b2d514b1f14a57c2d1343d/res/moves) 
- [PokeHeartGold/files/battledata/script/move_script/](https://github.com/pret/pokeheartgold/tree/ad7a3afa0cfc144fe6837c410cb95b2727217f54/files/battledata/script/move_script) 

### Battle Effect Scripts
**Battle effect scripts** determine the effect to execute when a move is used. For example, the move *Sleep Powder* has the effect of putting the target to sleep.

> Battle effect scripts may be shared by or assigned to multiple moves (e.g. both *Sleep Powder* and *Hypnosis* are assigned the same battle effect script that puts the target to sleep). **DSPRE's Move Editor** can change a move's assigned battle effect script (labeled as `Effect Sequence`, see [Basic Move Data](#basic-move-data)). 

Reviewer: @Mixone-FinallyHere would it be reasonable to change the label in DSPRE (at least the future Avalonia release) to be "Battle Effect Script" instead of "Effect Sequence"?

The actual logic of a battle effect script can be edited either by using a hex editor, by setting up a decomp project, or by setting up a [HG-Engine](https://github.com/BluRosie/hg-engine#about) project.

See [Battle Effects](#battle-effects) for more comprehensive details on how battle effect scripts work and guides on how to edit them.

### Battle Subscripts
**Battle subscripts** are helper scripts that contain logic supporting various aspects of the Gen IV battle system. When it comes to move execution, many *battle effect scripts* will point to a battle subscript that contains the logic to actually execute the overall battle effect (i.e. a move will execute its assigned battle effect script, which may have an instruction that points to a certain battle subscript to carry out the effect).

More comprehensive details on how battle subscripts work and guides on how to edit them are explored alongside battle effect scripts in the [Battle Effects](#battle-effects) section.

### Move Animation Scripts

> **Move animation scripts**, also known as move animations, contain commands to execute visual and audio effects when a move is used. Every move is uniquely associated with a move animation script. In other words, a uniquely identified move animation script is associated strictly with a specific move and may not shared.

Reviewer: ```suggestion
**Move animation scripts**, also known as move animations, contain commands to execute visual and audio effects when a move is used. Every move is uniquely associated with a move animation script. In other words, a uniquely identified move animation script is associated strictly with a specific move and may not be shared.
```

- [PokeHeartGold/files/battledata/script/effect_script](https://github.com/pret/pokeheartgold/tree/ad7a3afa0cfc144fe6837c410cb95b2727217f54/files/battledata/script/effect_script)

Below are the file locations of the **NARCs** containing battle effect scripts for each game:
| Game      | HeartGold/SoulSilver | Platinum                    | Diamond/Pearl               |
|-----------|:--------------------:|:---------------------------:|:---------------------------:|
| **File**  | `/a/0/3/0`           | `/battle/skill/be_seq.narc` | `/battle/skill/be_seq.narc` |

Battle effect scripts contain varying complexity, so the following sections will start by breaking down the most basic battle effect script before moving on to more complex ones, then ending with examples on how to edit them.

#### Finneon, Use Pound! (The Most Basic Battle Effect Script)

![Pound](resources/Pound_IV.png)

The most basic battle effect script is script `000` (used by Pound, Tackle, Vine Whip, Water Gun, etc.), which contains the following logic:
1. Determine whether the move is a critical hit.
2. Calculate the damage the move will inflict.
3. End the battle effect script execution.

> In other words, any move assigned with battle effect script `000` simply deals damage. Additionally, more complex battle effect scripts that will both deal damage and (potenially) apply an effect will include this logic in conjunction to other logic for applying a guaranteed or a chance of a stat change to the user or target. Got it memorized?

Reviewer: ```suggestion
In other words, any move assigned with battle effect script `000` simply deals damage. Additionally, more complex battle effect scripts that will both deal damage and (potenially) apply an effect will include this logic in conjunction to other logic for applying a guaranteed or a chance of a stat change to the user or target.
```

The last element, `End`, signals the battle engine to stop execution of this battle effect script.

#### Breaking Down Battle Effect Script `000` (Hex)
Taking a look at battle effect script `000` in a hex editor looks like this for each of the Gen IV games: 

```
// HeartGold/SoulSilver
26 00 00 00 0F 00 00 00 E0 00 00 00

// Platinum
26 00 00 00 0F 00 00 00 DE 00 00 00

// Diamond/Pearl
26 00 00 00 0F 00 00 00 DA 00 00 00
```

The first thing to note is that when viewed in a hex editor, battle effect scripts **do not** have bytes to indicate labels or the start of a function (i.e. the `_000:` when viewing the file in the decompilation projects). The implications of this will be explored in later sections when reviewing more complex battle effect scripts.

The second thing to note is that the instruction for ending the battle effect script has different hex values across the Gen IV games as listed below:
|                              | HeartGold/SoulSilver  |   Platinum    | Diamond/Pearl   |
|:----------------------------:|:---------------------:|:-------------:|:---------------:|
| `End` instruction Hex Values | `E0 00 00 00`         | `DE 00 00 00` | `DA 00 00 00`   |

> Now, every 4 bytes is either an instruction or a parameter for an instruction. Depending on the battle effect script's complexity, it may be easier to understand by first separating the hex values for each instruction. This requires cross-referencing with the associated file in the decompilation projects.

Reviewer: ```suggestion
Every 4 bytes is either an instruction or a parameter for an instruction. Depending on the battle effect script's complexity, it may be easier to understand by first separating the hex values for each instruction. This requires cross-referencing with the associated file in the decompilation projects.
```

For instance, the hex values for battle effect script `000` can broken down as the following (keeping in mind that battle effect script `000` only has instructions that don't use parameters):
- `26 00 00 00` - Critical hit determination
- `0F 00 00 00` - Damage calculation
- `E0 00 00 00` / `DE 00 00 00` / `DA 00 00 00` - Ends the battle effect script execution [HGSS / Plat / DP]

> Simple right? Now, moving on to battle effect scripts that actually apply effects! 

Reviewer: ```suggestion
Now, moving on to battle effect scripts that actually apply effects! 
```

  - Will usually occur 100% of the time, regardless of the move's assigned *Side Effect Probability*, as long as the move can successfully execute.
- Battle effects scripts that apply effects as an **indirect effect**:
  - Are usually assigned to *damage-dealing physical or special moves* and include the instructions for calculating critical hits or damage,
  - Will usually occur after a move successfully hits the target(s), with a few exceptions depending on other logic,
  - And are usually subject to the move's assigned *Side Effect Probability*, with a few exceptions depending on other logic.

Additional research and experimentation of battle effect scripts have indicated the following:
- The vanilla Gen IV games do not have any battle effect scripts that mix direct and indirect effects in a single script.
- Despite the first point, it is possible to create battle effect scripts that include both direct and indirect effects, with some quirks as to how they play out (see [Creating New Battle Effect Scripts](#creating-new-battle-effect-scripts)).
- If a battle effect script were to include and execute both direct and indirect effects...

>   - The *direct effect* will occur first, and if including the damage and critical hit calculation instructions, will occur before damage has been dealt.

Reviewer: ```suggestion
  - The *direct effect* will occur first, and if it includes the damage and critical hit calculation instructions, will occur before damage has been dealt.
```

>   - The *indirect effect* will occur second, and if including the damage and critical hit calculation instructions, will occur after damage has been dealt, as expected.

Reviewer: ```suggestion
  - The *indirect effect* will occur second, and if it includes the damage and critical hit calculation instructions, will occur after damage has been dealt, as expected.
```

| Cannot Prevent          | Unused?                                     | `MOVE_SIDE_EFFECT_CANNOT_PREVENT`          | `-- -- 00 08` |
| Probabilistic           | Seemingly a backup flag to ensure an indirect effect is boosted by Serene Grace <br/> Used by battle effect scripts `253` (Flare Blitz) & `262` (Volt Tackle) | `MOVE_SIDE_EFFECT_PROBABILISTIC` | `-- -- 00 04` |
| Check HP and Substitute | The effect is not applied if the move hits a substitute <br/> Used by battle effect scripts `042` (Bind, Fire Spin, etc) & `261` (Whirlpool) | `MOVE_SIDE_EFFECT_CHECK_HP_AND_SUBSTITUTE` | `-- -- 00 02` |
| Check Substitute        | The effect is not applied if the move hits a substitute <br/> Used by battle effect scripts `34` (Pay Day) & `105` (Covet, Thief) | `MOVE_SIDE_EFFECT_CHECK_SUBSTITUTE` | `-- -- 00 01` |
| Break Screens           | Removes Light Screen and Reflect from the target's side, even if immune to the move's typing <br/> Used by battle effect script `186` (Brick Break) | `MOVE_SIDE_EFFECT_BREAK_SCREENS`      | `-- -- 80 00` |

</details>

The next section will go over the "battle effect" parameter.

> ### Subscript Pointers (One Step Closer to Applying the Actual Effects)

Reviewer: ```suggestion
### Subscript Pointers
```

Reviewer: If applied would require the hyperlink changing as well.

```
// Battle Effect Script 011 - Assigned to Harden
_000:
    UpdateVar OPCODE_SET, BTLVAR_SIDE_EFFECT_FLAGS_DIRECT, MOVE_SIDE_EFFECT_TO_ATTACKER|[MOVE_SUBSCRIPT_PTR_DEFENSE_UP_1_STAGE]
    End 

// Battle Effect Script 138 - Assigned to Steel Wing
_000:
    UpdateVar OPCODE_SET, BTLVAR_SIDE_EFFECT_FLAGS_INDIRECT, MOVE_SIDE_EFFECT_TO_ATTACKER|[MOVE_SUBSCRIPT_PTR_DEFENSE_UP_1_STAGE]
    CalcCrit 
    CalcDamage 
    End 
```

```
// Battle Effect Script 011 - Assigned to Harden

   32 00 00 00   07 00 00 00   02 00 00 00   [10 00] 00 40
   DE 00 00 00 

// Battle Effect Script 138 - Assigned to Steel Wing

   32 00 00 00   07 00 00 00   03 00 00 00   [10 00] 00 40
   26 00 00 00
   0F 00 00 00
   DE 00 00 00
```
We can see that both battle effect scripts use the `MOVE_SUBSCRIPT_PTR_DEFENSE_UP_1_STAGE` or `10 00 -- --` subscript pointer, which then points to the `subscript_update_stat_stage` battle subscript (as seen [here](https://github.com/pret/pokeplatinum/blob/f9e031910823d49d0bcda2db40217e82b9c6def8/include/data/move_side_effect_subscripts.h#L23)) that will finally handle applying the battle effect of raising the Defense stat.

> There are over 100 subscript pointers (see the links above), so it's unlikely this guide will explicitly list them all, but some of the more notable ones will be mentioned in later sections. 

Reviewer: ```suggestion
There are over 100 subscript pointers (see the links above), so this guide will not explicitly list them all, but some of the more notable ones will be mentioned in later sections. 
```

> Last stop, battle subscripts!

Reviewer: ```suggestion
```

|:-----------|:---:|:--------:|:----------------:|:--------------------------------------------------:|:-----------------------:|
| **Value**  | 344 | Physical | Opponent or Ally | 262 - 1/3 damage recoil, may cause Paralysis (Hit) | 10                      |

#### Volt Tackle's Move Script
Because Volt Tackle's move ID is 344, its associated move script ID will also be 344. We can take a look at move script 344 either from [PokePlatinum](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/moves/volt_tackle/script.s) or by unpacking `/battle/skill/waza_seq.narc` and opening file `0334` in a hex editor.

```
//  Move Script 344 - Decompiled Code
_000:
    GoToEffectScript 

// Move Script 344 - Hex Values

   24 00 00 00
```
Nothing special here, it's just the instruction to signal the battle engine to start executing the battle effect script assigned to Volt Tackle's move data.

#### Battle Effect Script 262
Battle Effect Script 262 is expected to have the following effects:
1. Cause recoil damage to the user equal to 1/3 of the damage dealt to the target,

> 2. Possibly paralyze the target after a successful hit.

Reviewer: ```suggestion
2. Possibly paralyse the target after a successful hit.
```

Author: The European English versions of the Gen IV games kept the American English spelling of "paralyze" for in-game move descriptions.

> Here's the decompiled code and associated hex values (organized for your convenience) of battle effect script 262. You can also view it from [PokePlatinum](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/effects/effect_script_0262.s) or by unpacking `/battle/skill/be_seq.narc` and opening file `0262` in a hex editor.

Reviewer: ```suggestion
Here's the decompiled code and associated hex values (organised for your convenience) of battle effect script 262. You can also view it from [PokePlatinum](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/effects/effect_script_0262.s) or by unpacking `/battle/skill/be_seq.narc` and opening file `0262` in a hex editor.
```

```
// Battle Effect Script 262 - PokePlatinum Decompilation Code
_000:
    CheckAbility CHECK_NOT_HAVE, BTLSCR_ATTACKER, ABILITY_RECKLESS, _008
    UpdateVar OPCODE_SET, BTLVAR_POWER_MULTI, 12

_008:
    UpdateVar OPCODE_SET, BTLVAR_SIDE_EFFECT_FLAGS_INDIRECT, MOVE_SIDE_EFFECT_PROBABILISTIC|MOVE_SIDE_EFFECT_TO_DEFENDER|MOVE_SUBSCRIPT_PTR_RECOIL_1_3_CHANCE_TO_PARALYZE
    CalcCrit 
    CalcDamage 
    End 

> // Battle Effect Script 262 - Hex Values (Organized)

Reviewer: ```suggestion
// Battle Effect Script 262 - Hex Values (Organised)
```

| First               | `OPCODE_SET`         | `07 00 00 00` | Operation code to apply - Set a value (or variable) to the variable | 
| Second              | `BTLVAR_SIDE_EFFECT_FLAGS_INDIRECT`                            | `03 00 00 00` | Specified variable - The *indirect* battle effect variable |
| Third (target part) | `MOVE_SIDE_EFFECT_PROBABILISTIC\|MOVE_SIDE_EFFECT_TO_DEFENDER` | `-- -- 00 84` | Value (in this case the effect targeting flag) - Special "Probabilistic" flag (`0x04` in hex) + apply effect to the defender/target (`0x80` in hex) |
| Third (effect part)  | `MOVE_SUBSCRIPT_PTR_RECOIL_1_3_CHANCE_TO_PARALYZE`             | `89 00 -- --` | Value (in this case the subscript pointer) - Subscript pointer ID 137 or `0x89` in hex |

Because this is a damage-dealing move, the battle effect is designated as an *indirect effect*, as expected, meaning the effect will *potentially* occur after the move successful hits the target (and to reiterate, there are `CalcCrit` and `CalcDamage` instructions following this instruction to actually calculate and apply damage).

> When we take a look at the effect targeting flag, there are actually *two* flags. One, `MOVE_SIDE_EFFECT_TO_DEFENDER` (`-- -- 00 80` in hex), indicates that the battle effect should apply to the target, which makes sense given that one of Volt Tackle's effect is to potentially paralyze the target. The second flag, `MOVE_SIDE_EFFECT_PROBABILISTIC` (`-- -- 00 04` in hex), seems to some sort of failsafe flag to ensure that an indirect battle effect (in this case, potentially causing paralysis) is properly affected by the ability Serence Grace (see [here](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/src/battle/battle_lib.c#L1551)). As a fun fact, the `MOVE_SIDE_EFFECT_PROBABILISTIC` flag is only used by two battle effect scripts, `253` (used by Flare Blitz) and `262` (Volt Tackle).

Reviewer: ```suggestion
When we take a look at the effect targeting flag, there are actually *two* flags. One, `MOVE_SIDE_EFFECT_TO_DEFENDER` (`-- -- 00 80` in hex), indicates that the battle effect should apply to the target, which makes sense given that one of Volt Tackle's effect is to potentially paralyse the target. The second flag, `MOVE_SIDE_EFFECT_PROBABILISTIC` (`-- -- 00 04` in hex), seems to some sort of failsafe flag to ensure that an indirect battle effect (in this case, potentially causing paralysis) is properly affected by the ability Serence Grace (see [here](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/src/battle/battle_lib.c#L1551)). As a fun fact, the `MOVE_SIDE_EFFECT_PROBABILISTIC` flag is only used by two battle effect scripts, `253` (used by Flare Blitz) and `262` (Volt Tackle).
```

> And lastly, the part you probably care about the most, the *subscript pointer* that points to a *battle subscript* that actually contains the logic for executing Volt Tackles effect!

Reviewer: ```suggestion
And lastly, the *subscript pointer* that points to a *battle subscript* that actually contains the logic for executing Volt Tackle's effect!
```

This battle effect script specifies the following subscript pointer: `MOVE_SUBSCRIPT_PTR_RECOIL_1_3_CHANCE_TO_PARALYZE`. This is subscript pointer [ID 137](https://github.com/pret/pokeplatinum/blob/f9e031910823d49d0bcda2db40217e82b9c6def8/generated/battle_move_subscript_ptrs.txt#L138) (`0x89` in hex), which [points](https://github.com/pret/pokeplatinum/blob/f9e031910823d49d0bcda2db40217e82b9c6def8/include/data/move_side_effect_subscripts.h#L144) to `subscript_recoil_1_3_chance_to_paralyze`, or battle subscript `226`, the script that will handle the effects of applying the recoil damage to the user *and* (potentially) applying paralysis to the target, which will be covered in the next section.

> #### Battle Subscript 226 (but wait, there's more)

Reviewer: ```suggestion
#### Battle Subscript 226
```

Let's crack open battle subscript `226`. You can also view it from [PokePlatinum](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/subscripts/subscript_recoil_1_3_chance_to_paralyze.s) or by unpacking `/battle/skill/sub_seq.narc` and opening file `0226` in a hex editor.

```
// Battle Subscript `226` - PokePlatinum Decompilation Code
_000:
    Call BATTLE_SUBSCRIPT_RECOIL_1_3
    CompareVarToValue OPCODE_FLAG_NOT, BTLVAR_BATTLE_CTX_STATUS, SYSCTL_APPLY_SECONDARY_EFFECT, _008
    Call BATTLE_SUBSCRIPT_PARALYZE

_008:
    End 

> // Battle Subscript 226 - Hex Values (Organized)

Reviewer: ```suggestion
// Battle Subscript 226 - Hex Values (Organised)
```

   3C 00 00 00   93 00 00 00 
   20 00 00 00   05 00 00 00   06 00 00 00   00 00 40 00   02 00 00 00 
   3C 00 00 00   1F 00 00 00 
   
   DE 00 00 00
```

At first glance, there isn't much code in this battle subscript. However, there are `Call` instructions (`3C 00 00 00` in hex) to call and execute *two other battle subscripts*, one that handles applying recoil damage (specifically 1/3 recoil, `93 00 00 00` in hex or battle subscript 147), and another that handles applying paralysis (`1F 00 00 00` in hex or battle subscript 31). 

The first called battle subscript that handles applying recoil damage (specifically 1/3 recoil, see [here](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/subscripts/subscript_recoil_1_3.s)) proceeds as follows:
1. Check if the attacker's ability is either the Rock Head or Magic Guard, to which no recoil damage would be applied if so,
2. Retrieve the damage dealt to the target,

> 3. Divide the amount by 3,

Reviewer: ```suggestion
3. Divide the amount by three,
```

4. Apply the resulting amount as recoil damage to the attacker, 
5. Display the recoil message,
6. And finally end, which in this case would return to the original battle subscript.

Afterwards, we have a conditional instruction, `CompareVarToValue`, that checks the `TRUE/FALSE` result of the `BattleSystem_TriggerSecondaryEffect` function (see [here](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/src/battle/battle_lib.c#L1519)). In layman terms, this checks whether or not the other effect of causing paralysis should occur, given that the move did successfully hit the target, and is subject to the move's assigned effect probability.

> If said check successfully passed, then this battle subscript calls another battle subscript that goes through a plethora of other checks (such as checking for Limber, Shield Dust, Harsh Sunlight weather + Leaf Guard, etc.) to determine if the target of the effect can actually be paralyzed (see [here](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/subscripts/subscript_paralyze.s)).

Reviewer: ```suggestion
If said check successfully passed, then this battle subscript calls another battle subscript that goes through a plethora of other checks (such as checking for Limber, Shield Dust, Harsh Sunlight weather + Leaf Guard, etc.) to determine if the target of the effect can actually be paralysed (see [here](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/subscripts/subscript_paralyze.s)).
```

The process to fix these battle effect scripts is relatively straightforward, involving copying the structure of other battle effect scripts with similar intended functionalities and substituting in the correct parameters. This also serves as a guided entry point in understanding how to create custom battle effect scripts.
:::

#### Battle Effect Script 012 Isn't Real, It Can't Hurt You
Let's look at battle effect script `012`, which is intended to increase the user's Speed stat by one stage. However, taking a look at the decompiled code (see [here](https://github.com/pret/pokeplatinum/blob/8b6fa504086925c957b2d514b1f14a57c2d1343d/res/battle/scripts/effects/effect_script_0012.s)) or by opening battle effect script `012` in a hex editor (unpack `/battle/skill/be_seq.narc` or `/a/0/3/0` and open file `0012`) reveals the following:

```
// Battle Effect Script 012 - PokePlatinum Decompilation Code
_000:
    CalcCrit 
    CalcDamage 
    End 

// Battle Effect Script 012 - Hex Values (Platinum)

   26 00 00 00 
   0F 00 00 00 
   DE 00 00 00
```

> Hold up, doesn't this look exactly like [battle effect script `000`](#inspecting-the-actual-data-of-battle-effect-script-000)? (Keeping in mind that the `End` instruction has different hex values across each Gen IV game, see [Breaking Down Battle Effect Script `000` (Hex)](#breaking-down-battle-effect-script-000-hex)). Regardless, this means that battle effect script `012` currently contains only the basic (damaging) move effects of determining critical hits and calculating damage. This is also observed with the other "broken" battle effect scripts, which will be listed later (and again, labeled with a `Dummy` prefix in DSPRE's Move Editor).

Reviewer: ```suggestion
This looks exactly like [battle effect script `000`](#inspecting-the-actual-data-of-battle-effect-script-000) (keeping in mind that the `End` instruction has different hex values across each Gen IV game, see [Breaking Down Battle Effect Script `000` (Hex)](#breaking-down-battle-effect-script-000-hex)). Regardless, this means that battle effect script `012` currently contains only the basic (damaging) move effects of determining critical hits and calculating damage. This is also observed with the other "broken" battle effect scripts, which will be listed later (and again, labelled with a `Dummy` prefix in DSPRE's Move Editor).
```

> #### "Broken" Battle Effect Scripts, We Can Fix Them

Reviewer: ```suggestion
#### "Broken" Battle Effect Scripts
```

  </tr>
  <tr>
    <td>`74`</td>

>     <td>Reduce the target's Evasion stat by one stage (damaging indirect effect).</td>

Reviewer: The original content is accurate. Reduce evasion on hit (74) is the last in the chain of battle effects 68-73. It is fair that 68-69 appear to use a different sequence (the pattern I think you've used), but the majority (70-73) use the pattern originally described.

Author: I prefer this pattern because most scripts for moves that deal damage and have indirect effects put the `26 00 00 00 0F 00 00 00` / `CalcCrit CalcDamage` at the end of the script (there usually isn't a functional difference though).

- `141`

> - `157`

Reviewer: ```suggestion
- `157` (*Unused, but does have the same trainer move selection AI routines associated to it as healing moves such as Synthesis*)
```

- `163`
- `264` (*Identified as "Unused" in DSPRE, but has [unique code](https://github.com/pret/pokeplatinum/blob/b2cd286f3d431cf6226c55139b58e6b140a5c827/res/battle/scripts/effects/effect_script_0264.s).*)
- `277` - `470` (*Identified as "Undocumented" in DSPRE and do not exist as a file in the battle effect scripts NARC*)
  

> The safest IDs to repurpose are those in the `277`-`470` range. Since these files do not exist at all, they are less likely to have some unknown or unclear purpose or reference in the game's code. 

Reviewer: ```suggestion
The safest IDs to repurpose are those in the `277`-`470` range. Since these files do not exist at all, they are less likely to have some unknown or unclear purpose or reference in the game's code.  
However, IDs `133`, `134` and `157` may be advantageous to use in conjunction with re-directing their trainer move selection AI routines (Basic and Expert flags) to already existing routines or new ones, as it does not require adding new entries to the AI flags' `main` routine.
```

### Creating New Battle Effect Scripts
> Source(s): [Drayano](https://pastebin.com/u/DrayHackTutorials), Yako, Lmaokai, HG-Engine  

#### Drayano's Tutorial
A [tutorial](https://pastebin.com/a5bGatsc) written by Drayano, details how to utilise the (possibly unintended) ability of declaring both direct and indirect effects in a single battle effect script to create stat-changing status moves that do not exist in the Gen IV games (such as Coil or modern Growth effects).  

> Because this method uses only existing battle subscripts, the appearance is not as clean. For example, a creating Coil approximation through this method could utilise the direct effect to raise Attack & Defense, and then utilise the indirect effect to raise Accuracy. However, this results in *two* stat-increase animations, but this is also one of the more accessible methods of "backporting" stat-changing moves from later Pokémon games.

Reviewer: ```suggestion
Because this method uses only existing battle subscripts, the appearance is not as clean. For example, a creating Coil approximation through this method could utilise the direct effect to raise Attack & Defense, and then utilise the indirect effect to raise Accuracy. This results in *two* stat-increase animations, but it is also one of the more accessible methods of "backporting" stat-changing moves from later Pokémon games.
```

> The following are the summarized steps for creating new battle effect scripts (based on Drayano's tutorial):

Reviewer: ```suggestion
The following are the summarised steps for creating new battle effect scripts (based on Drayano's tutorial):
```

1. Unpack the battle effect script NARC from your project's DSPRE_contents folder.
2. Make a copy of file `0000`, and rename it to the next available file ID number (for example `0277` for any Gen IV game if this is the first new battle effect script you're creating).
3. Open the new file in a hex editor and delete the contents.

> 4. ~~Draw the owl~~ Make your effect! Reference Drayano's tutorial and the following sections: [Direct and Indirect Effects](#direct-and-indirect-effects), [Effect Targeting Flag](#effect-targeting-flag), and [Subscript Pointers](#subscript-pointers-one-step-closer-to-applying-the-actual-effects).

Reviewer: ```suggestion
4. Make your effect! Reference Drayano's tutorial and the following sections: [Direct and Indirect Effects](#direct-and-indirect-effects), [Effect Targeting Flag](#effect-targeting-flag), and [Subscript Pointers](#subscript-pointers-one-step-closer-to-applying-the-actual-effects).
```

</details>

### Creating New Battle Subscripts
It is possible to create **new battle subscripts** to accomodate more complex stat changes and other battle effects not present in the Gen IV games, but adding new battle subscripts involves two major requirements:
1. Creating new battle subscript files and adding them to the battle subscript NARC (similar process to adding new battle effect scripts).
2. Moving the **subscript pointer table** and adding entries to it.

#### The Bottleneck: The Subscript Pointer Table
As mentioned in an earlier section, *subscript pointers* are pointers to a certain *battle subscript* that contains the logic to actually apply a battle effect. This mapping of subscript pointers and battle subscripts is stored in the overlay file that handles battle logic for each Gen IV game. However, creating new battle subscripts will *usually* require adding an entry to the table, to which there is simply no more space for additional entries. Thus, the table needs to be moved to in order to accomodate additional entries.

> But before detailing how to move the subscript pointer table, let's discuss more information about it first.

Reviewer: ```suggestion
```

3. Save the overlay file.

**Step 4: Save and test** <br/>
1. Save your project, open your game, and test moves with different battle effects to verify the game does not crash.
2. If the game does not crash after testing different kinds of moves, then the subscript pointer table has been successfully moved.

#### Steps for Creating New Battle Subscripts
Creating new battle subscripts which handle different combinations of two or more 1- or 2-stage stat changes is achievable by mirroring battle subscripts for existing effects such as [Calm Mind](https://github.com/pret/pokeheartgold/blob/master/files/battledata/script/subscript/subscript_0151_CalmMind.s), [Dragon Dance](https://github.com/pret/pokeheartgold/blob/master/files/battledata/script/subscript/subscript_0152_DragonDance.s) and [Curse (non-Ghost)](https://github.com/pret/pokeheartgold/blob/master/files/battledata/script/subscript/subscript_0096_CurseNormal.s). 

> The following are the summarized steps for creating new battle subscripts:

Reviewer: ```suggestion
The following are the summarised steps for creating new battle subscripts:
```

1. Unpack the battle subscript NARC from your project's DSPRE_contents folder.
2. Make a copy of file `0000`, and rename it to the next available file ID number (for example `0297` for HeartGold/SoulSilver/Platinum or `0293` for Diamond/Pearl if this is the first new battle subscript).
3. Open the new file in a hex editor and delete the contents.

> 4. ~~Draw the owl, again~~ Make your new battle subscript! Reference the [Battle Subscripts](#battle-subscripts-1) and [Example Case Studies](#example-case-studies) sections.

Reviewer: ```suggestion
4. Make your new battle subscript! Reference the [Battle Subscripts](#battle-subscripts-1) and [Example Case Studies](#example-case-studies) sections.
```

- [List of battle variables](https://github.com/pret/pokeplatinum/blob/6e321e13e2c155fa79185a6366a7846cd904f808/generated/battle_script_vars.txt) - For example, the `Power Multiplier` variable.
- [List of battler characteristics](https://github.com/pret/pokeplatinum/blob/6e321e13e2c155fa79185a6366a7846cd904f808/generated/battle_mon_params.txt) - For example, a battler's current `Attack_Stage`.
- [List of battle message tags/string buffers](https://github.com/pret/pokeplatinum/blob/6e321e13e2c155fa79185a6366a7846cd904f808/generated/battle_message_tags.txt) - For example, `{Pokémon_Nickname} was badly poisoned!`.

</details>

------------------------------

## Move Animations  
> Source(s): Acent, [DavveDP](https://github.com/DavveDP), [Fexty](https://github.com/Fexty12573). [HG-Engine](https://github.com/BluRosie/hg-engine/wiki/Move-Animation-Scripting-System-Documentation)

**Move animations** are the displayed visual (and audio) effects when a battler uses a move. Move animations are executed by *move animation scripts* that call and manipulate various components such as particle emitters, sprite translations, background effects, and sound effects. 

> There are four main types of components utilized by move animation scripts:

Reviewer: ```suggestion
There are four main types of components utilised by move animation scripts:
```

    - [Original Release (Discord link)](https://discord.com/channels/446824489045721090/482242795982159872/799743818797940797)
- [WazaEffectEditor (HeartGold - Discord link)](https://discord.com/channels/446824489045721090/534415767454875678/1250447089067102300)
    - [GitHub Repo](https://github.com/DavveDP/Waza-Editor)

The first time that WazaEffectEditor is opened a message about downloading libraries/moves from Bulbapedia may be presented, this should be accepted. This may then result in a hanging state after a few minutes. If this is the case, closing and re-opening the application should resolve and allow use of the tool.

WazaEffectEditor works on the same premise as tools like DSPRE, i.e. the user selects a ROM (`.nds` file) or project folder, and WazaEffectEditor unpacks it (using the ROM's move names and animations). When saving, the ROM or project folder is re-packed and saved (with the option to override the existing ROM file, or rename). It is important to be aware of this if edits are being done across multiple tools, as the HeartGold version of WazaEffectEditor doesn't interact with the unpacked contents folder that DSPRE creates.

:::warning

> WazaEffectEditor may not be compatible with projects using **DSPRE versions 2.0+** due to file structure changes. In this case, WazaEffectEditor can either be used as a reference while directly hex editing move animation scripts, or be used with a separate project created with an older DSPRE version and copying over the move animation script NARC after edits have been made. 

Reviewer: As far as I know, this isn't a concern. When you open a ROM in WEE it uses it's own included ndstool to unpack the ROM into a temporary `_ext` folder in WEE's own application folder (this is on the HG version, it's possible something is different with the Plat version, but I don't believe so?)

Author: The one I have for Platinum only allows for opening a SDSME folder

Reviewer: Oh, noted. Must be one of the differences between the two versions then I guess

Reviewer: Maybe: "Depending on the version used, WazaEffectEditor may not be compatible with...." or something to that effect?

Author: Oh actually I just realized the HeartGold version of WEE can also open Platinum as well, I'll base the section on that version

Below are the file locations of the **NARCs** containing the particle animation systems for each game:
| Game      | HeartGold/SoulSilver | Platinum/Diamond/Pearl                      |
|-----------|:--------------------:|:-------------------------------------------:|
| **File**  | `/a/0/2/9`           | `/wazaeffect/effectdata/waza_particle.narc` |

[NitroEFX](https://github.com/Fexty12573/nitroefx) is a tool created by [Fexty](https://github.com/Fexty12573) for viewing and editing these particle animation systems.

When opening NitroEFX, select *File > Open > SPL File* and open your project's respective particle animation system file. You can select any `.bin` file in the top left panel, then select a component (the specific "particle emitter") in the bottom left panel, and in the top right panel choose either the *Play All Emitters* option or the *Play Emitter* option (which plays only the selected component).

:::info
For **HeartGold/SoulSilver**, `/a/0/2/9` needs to be temporarily renamed to `/a/0/2/9.narc` in order to be opened in NitroEFX
:::

As an example, let's take a quick look at Karate Chop's move animation and the particle emitters it uses in *Pokémon Platinum*. 

> First, we'll need to use either WazaEditor or a hex editor to identify the particle animation system that the move animation script loads (unfortunately, the PokePlatinum decompiled code uses an actual name instead of an ID, which isn't helpful in this case). In WazaEditor, the `SetPlayAnm 0x0 0x20 0x1` command indicates that particle animation system number `0x20` (`32` in decimal) is used. 

Reviewer: ```suggestion
First, we'll need to use either WazaEffectEditor or a hex editor to identify the particle animation system that the move animation script loads (unfortunately, the PokePlatinum decompiled code uses an actual name instead of an ID, which isn't helpful in this case). In WazaEffectEditor, the `SetPlayAnm 0x0 0x20 0x1` command indicates that particle animation system number `0x20` (`32` in decimal) is used. 
```

> Next, use NitroEFX to open `/wazaeffect/effectdata/waza_particle.narc` and select `32.bin` from the top left panel. In the bottom left panel, we can see three components, or particle emitters, contained in this particle animation system. These three components are called in the move animation script, as seen by the `LoadAnm 0x0 0x2 0x4`, `LoadAnm 0x0 0x0 0x4`, and `LoadAnm 0x0 0x1 0x4` commands in WazaEditor (keeping in mind that the first parameter of the `LoadAnm` command is the local ID of the currently loaded animation particle system(s) and the second parameter is the specific particle emitter to play). 

Reviewer: ```suggestion
Next, use NitroEFX to open `/wazaeffect/effectdata/waza_particle.narc` and select `32.bin` from the top left panel. In the bottom left panel, we can see three components, or particle emitters, contained in this particle animation system. These three components are called in the move animation script, as seen by the `LoadAnim 0x0 0x2 0x4`, `LoadAnim 0x0 0x0 0x4`, and `LoadAnim 0x0 0x1 0x4` commands in WazaEffectEditor (keeping in mind that the first parameter of the `LoadAnim` command is the local ID of the currently loaded animation particle system(s) and the second parameter is the specific particle emitter to play). 
```

Select a specific particle emitter. The top right panel should now have the options to *Play Emitter* or *Play All Emitters*. Note that the *Play All Emitters* option may not necessarily play the emitters in the same sequence or timing as the actual move animation, as that is instead typically managed by the order of commands in the move animation script.

> For example, if you play the second particle emitter (ID of `[1]`), you'll see that it is specifically the "chopping hand" component of this whole animation. As such, this process may be helpful in identifying specific components of an animation to utilize when creating new move animation scripts.

Reviewer: ```suggestion
For example, if you play the second particle emitter (ID of `[1]`), you'll see that it is specifically the "chopping hand" component of this whole animation. As such, this process may be helpful in identifying specific components of an animation to utilise when creating new move animation scripts.
```

- **Nintendo Unity (Youtube) - [Generation I-VII](https://www.youtube.com/playlist?list=PLsOPwXA-m0-Bk-3P3avQGg-poPESOibEw) and [Generation IX](https://www.youtube.com/watch?v=B01cWKbAEw4) Move Animations**

</details>

<details>
<summary>The following resources are from the PokePlatinum decompilation project and may be helpful for cross-referencing commands and parameters:</summary>

- Decompiled code of each [move animation script](https://github.com/pret/pokeplatinum/tree/6e321e13e2c155fa79185a6366a7846cd904f808/res/moves)
- [Indexed list](https://github.com/Fexty12573/pokeplatinum/blob/13213bf0417feef5e058d4e251d24e10bbe8bc9d/src/battle_anim/battle_anim_system.c#L790) of commands that can be used in move animation scripts
- Move animation [command parameters](https://github.com/pret/pokeplatinum/blob/6e321e13e2c155fa79185a6366a7846cd904f808/include/constants/battle/battle_anim.h), including:
   - Background transition types
   - Battler sprite targets (to apply animation particle system visual elements to)
   - Battler sprite targets (for other various functions to apply their effect to)
   - Battler sprite translation and movement types

>    - Available colors (i.e. the color to fade a background or sprite to)

Reviewer: ```suggestion
   - Available colours (i.e. the colour to fade a background or sprite to)
```
